x-pack/plugins/security_solution/public/timelines/components/timeline/body/actions/index.tsx

```diff
@@ -3,203 +3,90 @@
+              <EuiCheckbox
+                data-test-subj="select-event"
                 id={eventId}
-                onClick={onEventToggled}
+                checked={checked}
+                onChange={handleSelectEvent}
               />
             )}
           </EventsTdContent>
         </EventsTd>
+      )}
+      <EventsTd key="expand-event">
+        <EventsTdContent textAlign="center" width={DEFAULT_ICON_BUTTON_WIDTH}>
+          {loading ? (
+            <EventsLoading />
+          ) : (
+            <EuiButtonIcon
+              aria-label={expanded ? i18n.COLLAPSE : i18n.EXPAND}
+              data-test-subj="expand-event"
+              iconType={expanded ? 'arrowDown' : 'arrowRight'}
+              id={eventId}
+              onClick={onEventToggled}
+            />
+          )}
+        </EventsTdContent>
+      </EventsTd>
 
-        <>{additionalActions}</>
+      <>{additionalActions}</>
+    </EventsTdGroupActions>
+  );
+};
 
-        {!isEventViewer && (
-          <>
-            <EventsTd key="timeline-action-pin-tool-tip">
-              <EventsTdContent textAlign="center" width={DEFAULT_ICON_BUTTON_WIDTH}>
-                <EuiToolTip
-                  data-test-subj="timeline-action-pin-tool-tip"
-                  content={getPinTooltip({
-                    isPinned: eventIsPinned,
-                    eventHasNotes: eventHasNotes(noteIds),
-                    timelineType: timeline.timelineType,
-                  })}
-                >
-                  <Pin
-                    allowUnpinning={!eventHasNotes(noteIds)}
-                    data-test-subj="pin-event"
-                    onClick={onPinClicked}
-                    pinned={eventIsPinned}
-                    timelineType={timeline.timelineType}
-                  />
-                </EuiToolTip>
-              </EventsTdContent>
-            </EventsTd>
+ActionsComponent.displayName = 'ActionsComponent';
 
-            <EventsTd key="add-note">
-              <EventsTdContent textAlign="center" width={DEFAULT_ICON_BUTTON_WIDTH}>
-                <NotesButton
-                  animate={false}
-                  associateNote={associateNote}
-                  data-test-subj="add-note"
-                  getNotesByIds={getNotesByIds}
-                  noteIds={noteIds || emptyNotes}
-                  showNotes={showNotes}
-                  size="s"
-                  status={timeline.status}
-                  timelineType={timeline.timelineType}
-                  toggleShowNotes={toggleShowNotes}
-                  toolTip={
-                    timeline.timelineType === TimelineType.template
-                      ? i18n.NOTES_DISABLE_TOOLTIP
-                      : i18n.NOTES_TOOLTIP
-                  }
-                  updateNote={updateNote}
-                />
-              </EventsTdContent>
-            </EventsTd>
-          </>
-        )}
-      </EventsTdGroupActions>
-    );
-  },
-  (nextProps, prevProps) => {
```

Comment:
Thanks for removing this touchpoint, which can be the source of subtle bugs. No action is required for this PR, but it probably makes sense to (manually) test for unnecessary re-renders after the timeline query has been migrated to the "search strategy".